File: gcClient/src/app/models/Jewelry.ts


interface Jewelry extends Product{
    weight: number;
    forGender: number;
    diamondCarat: number;
    // sizes: string[],
    // quantities: number[];
    materials: string[];
    gemstones: Gemstone[];
}

interface JewelryParams extends ProductParams {
    forGender: number;
    materials: string[];
    gemstones: string[]; 
  }

// interface Ring extends Jewelry {
//     availableSizes: number[];
//   }

File: GoldStore/GoldStoreAPI/Entyties/Products/Jewelleries/Jewelry.cs
using System.Text.Json.Serialization;

namespace GoldStoreAPI.Entyties.Products.Jewelleries
{
    public class Jewelry : Product
    {
        public double Weight { get; set; }

        [JsonConverter(typeof(JsonStringEnumConverter))]
        public ForGender ForGender { get; set; }

        public double? DiamondCarat { get; set; }

        //public string JewelryType { get; set; }

        //public float[]? AvailableSizes { get; set; }

        public List<Material>? Materials { get; set; }

        public List<Gemstone>? Gemstones { get; set; }

        //public string[] Sizes { get; set; }

        //public int[] Quantities { get; set; }
        //public override int QuantityInStock { get => Quantities[Array.FindIndex(Sizes, x => x == base.Size)]; set => Quantities[Array.FindIndex(Sizes, x => x == base.Size)] = value; }

    }


}
